test/integration/ctl/ctl_create_cr_test.go

```diff
@@ -288,31 +464,31 @@ func TestCtlCreateCR(t *testing.T) {
 				t.Errorf("CR created in unexpected Namespace, expected: %s, actual: %s", test.expNamespace, gotCr.Namespace)
 			}
 
-			// Check the file where the private key is stored
-			keyData, err := ioutil.ReadFile(test.expKeyFilename)
-			if err != nil {
-				t.Errorf("error when reading file storing private key: %v", err)
-			}
-			_, err = pki.DecodePrivateKeyBytes(keyData)
-			if err != nil {
-				t.Errorf("invalid private key: %v", err)
-			}
-
 			// If applicable, check the file where the certificate is stored
-			if test.fetchCert {
+			// If the expected error message is the one below, we skip checking
+			// because no certificate will have been written to file
+			if test.fetchCert && test.expErrMsg != "error when waiting for CertificateRequest to be signed: timed out waiting for the condition" {
 				certData, err := ioutil.ReadFile(test.expCertFilename)
 				if err != nil {
 					t.Errorf("error when reading file storing private key: %v", err)
 				}
 
 				if !bytes.Equal(test.expCertFileContent, certData) {
-					t.Errorf("certificate written to file is wrong, expected: %v,\nactual: %v", test.expCertFileContent, certData)
+					t.Errorf("certificate written to file is wrong, expected: %s,\nactual: %s", test.expCertFileContent, certData)
 				}
 			}
 		})
 	}
 }
 
+func getTestDataDir(t *testing.T) string {
+	testWorkingDirectory, err := os.Getwd()
+	if err != nil {
+		t.Fatal(err)
+	}
+	return testWorkingDirectory + "/testdata/"
+}
+
```

Comment:
I'm not sure if we need this. `go test` will automatically set the working-dir to match the package path under test, so `testdata/` can always be reached in tests using that relative path. See https://dave.cheney.net/2016/05/10/test-fixtures-in-go

Reply:
We need this for this test because we cd into a tmp directory for each test for easier clean-up of the generated files.

Reply:
Is this a tmp directory?

Reply:
The `testdata` dir is not a tmp directory, but the working directory for each test is a tmp directopry that is set up by `setupPathForTest`